oshi-core/src/main/java/oshi/util/ExecutingCommand.java

```diff
@@ -116,16 +116,39 @@ private ExecutingCommand() {
      *         string if the command failed
      */
     public static List<String> runNative(String[] cmdToRunWithArgs, String[] envp) {
+        Process p = null;
         try {
-            Process p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
-            return getProcessOutputAndDestroy(p, cmdToRunWithArgs);
+            p = Runtime.getRuntime().exec(cmdToRunWithArgs, envp);
+            return getProcessOutput(p, cmdToRunWithArgs);
         } catch (SecurityException | IOException e) {
             LOG.trace("Couldn't run command {}: {}", Arrays.toString(cmdToRunWithArgs), e.getMessage());
+        } finally {
+            if (p != null) {
+                p.destroy();
```

Comment:
OK, I've dug into the JDK8 source code enough to see that Solaris is indeed handled a little differently. The JDK8 `destroy()` method for Solaris simply executes `kill`.  In fact there's a separate test case `StreamsSurviveDestroy` that demonstrates this behavior and [indicates it's unique to Solaris](https://github.com/openjdk/jdk8/blob/6a383433a9f4661a96a90b2a4c7b5b9a85720031/jdk/test/java/lang/Runtime/exec/StreamsSurviveDestroy.java#L191):
```
// Applies only to Solaris; Linux and Windows
// behave a little differently
if (!System.getProperty("os.name").equals("SunOS"))
    return;
```
So given, this, please wrap the below stream closing conditionals inside a check for the Solaris platform.  For our purposes, that `os.name` match is done in JNA when determining the Platform type so you could just do 
```java
if Platform.isSolaris() { 
    // ... close streams ... 
}
```

Reply:
I'm not sure if this behavior occurs on AIX. It may be preferred to _exclude_ known-good (JDK-supported) OS's: Linux, Windows, macOS, and BSD unix, and permit the checks on any others (Solaris and AIX).